src/lib/components/repo/RepoReleases.svelte

```diff
@@ -31,13 +31,22 @@
   let rateLimitMessage = $state<string | null>(null);
   let rateLimitHitOnFirst = $state(false);
 
+  // Filter options
+  let filterText = $state('');
+
   // Sorting options
   let sortBy = $state<'name' | 'count' | 'date'>('name');
   let sortOrder = $state<'asc' | 'desc'>('asc');
-  let sortedGroups = $state<PackageGroup[]>([]);
-
-  // Filter options
-  let filterText = $state('');
+  let sortedGroups: PackageGroup[] = $derived(
+    (() => {
+      if (!groupedReleases) return [];
+      const ft = filterText.toLowerCase();
+      const needFilter = !!filterText;
+      return sortPackageGroups(groupedReleases.groups, sortBy, sortOrder).filter(
+        (g) => !needFilter || g.name.toLowerCase().includes(ft),
+      );
+    })(),
+  );
```

Comment:
```suggestion
  let sortedGroups: PackageGroup[] = $derived.by(() => {
    if (!groupedReleases) return [];

    let groups = groupedReleases.groups;
    
    if (!!filterText) {
      const searchTerm = filterText.toLowerCase();
      groups = groups.filter(group =>
        group.name.toLowerCase().includes(searchTerm)
      );
    }
   
    return sortPackageGroups(groups, sortBy, sortOrder)
  });
```

This is how I suggest doing it.